sources/xfscli/src/bin/fsck_xfs.rs

```diff
@@ -0,0 +1,199 @@
+        return Err(Error::DeviceNotFound {
+            target: target.to_string(),
+        });
+    }
+
+    if force {
+        // Preform xfs_repair
+        println!("Performing XFS repair..");
+        let mut repair_exit_code = perform_xfs_repair(vec!["-e", target]);
+
+        // Clear the log, mount and unmount the XFS file system
+        // before xfs_repair
+        if repair_exit_code == XfsRepairResponseCode::DirtyLogs && repair {
+            println!("Replaying log for {}", target);
+
+            // Using random directory (/tmp/repair_mntxxxx) instead of a targeted directory (/tmp/repair_mnt)
+            let mnt_dir = tempfile::Builder::new()
+                .prefix(REPAIR_MOUNT_DIR)
+                .tempdir()
+                .context(error::MakeDirSnafu)?;
+
+            let mnt_dir_path = mnt_dir
+                .path()
+                .to_str()
+                .expect("Unable to get path of temporary dir");
+
+            let kernel_params = get_root_and_rootflags(target);
+
+            let basename_dev = Path::new(target)
+                .file_name()
+                .expect("Unable to find device base name");
+
+            let basename_root = Path::new(&kernel_params.root)
+                .file_name()
+                .expect("Unable to find root base name");
+
+            let mut mount_args = vec![target.as_str(), mnt_dir_path];
+            if basename_dev == basename_root && !kernel_params.root_flags.is_empty() {
+                mount_args.push(kernel_params.root_flags.as_str());
+            }
+
+            let mount_status = Command::new(MOUNT)
+                .args(mount_args)
+                .status()
+                .expect("Couldn't run mount");
+
+            if !mount_status.success() {
+                return Err(Error::Mount);
+            }
+
+            Command::new(UMOUNT)
+                .arg(mnt_dir_path)
+                .output()
+                .expect("Couldn't run unmount");
+
+            println!("Performing XFS repair again after cleaning the log..");
+            repair_exit_code = perform_xfs_repair(vec!["-e", target]);
+
+            let mnt_dir_close_msg = format!("Could not delete the {} dir", mnt_dir_path);
+            mnt_dir.close().expect(&mnt_dir_close_msg);
+        }
+
+        let metadata_repaired = match repair_exit_code {
+            XfsRepairResponseCode::Ok => Ok(false),
+            XfsRepairResponseCode::RepairFailure => Err(Error::RepairFailure),
+            XfsRepairResponseCode::DirtyLogs => Err(Error::DirtyLogs),
+            XfsRepairResponseCode::MetadataRepair => Ok(true),
+            XfsRepairResponseCode::CommandUnavailable => Err(Error::CommandUnavailable {
+                cli: "fsck".to_string(),
+                command: "xfs_repair".to_string(),
+            }),
+            XfsRepairResponseCode::Unknown(code) => Err(Error::UnrecognizedExitCode { code }),
+        }?;
+
+        return Ok(if metadata_repaired { 1 } else { 0 });
```

Comment:
If you do move to using `ExitStatus`, this could be a good spot to use `ExitCode::SUCCESS` and `ExitCode::FAILURE`.

Reply:
We need to return 1 as success code following https://git.kernel.org/pub/scm/fs/xfs/xfsprogs-dev.git/tree/fsck/xfs_fsck.sh, so I am not sure if we should return as ExitCode::FAILURE.

Reply:
Ah wow, I wasn't expecting that interface.

I'd personally still prefer to return `ExitCode`, but manually provide the value yourself, e.g.

```rust
ExitCode::from(1)
```

Reply:
Changed all the i32 return types to ExitCode and return Exitcode from main function.